box/box_test.go

```diff
@@ -27,3 +31,19 @@ func TestNew(t *testing.T) {
 		_, _ = b.Info()
 	})
 }
+
+func TestNew_Schema_PassesDoer(t *testing.T) {
+	t.Parallel()
+
+	doer := test_helpers.NewMockDoer(t)
+
+	b := New(&doer)
+	schemaObject := b.Schema()
+	assert.Equal(t, &doer, schemaObject.conn)
+
+	userSchemaObject := b.Schema().User()
+	assert.Equal(t, &doer, userSchemaObject.conn)
+
+	sessionObject := b.Session()
+	assert.Equal(t, &doer, sessionObject.conn)
```

Comment:
You could create a mock of the Doer interface here and make sure that the methods are called exactly from the mock instead of sharing the internal implementation.